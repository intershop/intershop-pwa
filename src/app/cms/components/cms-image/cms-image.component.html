<ng-container *ngIf="pagelet && pagelet.hasParam('Link'); else image">
  <a [routerLink]="routerLink(pagelet, 'Link')"
    [attr.title]="pagelet.stringParam('LinkTitle')"
    class="teaser-link"
  >
    <ng-container *ngTemplateOutlet="image"></ng-container>
  </a>
</ng-container>

<ng-template #image>
  <img [src]="pagelet.stringParam('Image')"
    [attr.alt]="pagelet.stringParam('AlternateText')"
    [ngClass]="pagelet.stringParam('CSSClass')"
  >
</ng-template>

<!-- TODO: (CMS Component Configuration)
  Picture
-->
