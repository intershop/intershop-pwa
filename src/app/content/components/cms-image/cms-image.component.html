<ng-container *ngIf="getConfigurationParameterValue(pagelet, 'Link'); else image">
  <a [routerLink]="getRouterLink(getConfigurationParameterValue(pagelet, 'Link'))"
    [attr.title]="getConfigurationParameterValue(pagelet, 'LinkTitle')"
    class="teaser-link"
  >
    <ng-container *ngTemplateOutlet="image"></ng-container>
  </a>
</ng-container>

<ng-template #image>
  <img [src]="getImagePath(getConfigurationParameterValue(pagelet, 'Image'))"
    [attr.alt]="getConfigurationParameterValue(pagelet, 'AlternateText')"
    [ngClass]="getConfigurationParameterValue(pagelet, 'CSSClass')"
  >
</ng-template>

<!-- TODO: (CMS Component Configuration)
  Picture
-->
