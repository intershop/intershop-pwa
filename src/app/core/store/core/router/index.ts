// API to access ngrx core router state
export * from './router.selectors';
export * from './router.operators';
