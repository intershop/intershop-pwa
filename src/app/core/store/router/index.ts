export * from './router.actions';
export * from './router.selectors';
export { CoreState } from '../core.state';
