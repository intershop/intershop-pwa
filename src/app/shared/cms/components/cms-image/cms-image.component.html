<ng-container *ngIf="pagelet && pagelet.hasParam('Link'); else image">
  <ng-container *ngIf="isRouterLink(pagelet, 'Link'); else externalLink">
    <a [routerLink]="routerLink(pagelet, 'Link')" [attr.title]="pagelet.stringParam('LinkTitle')" class="teaser-link">
      <ng-container *ngTemplateOutlet="image"></ng-container>
    </a>
  </ng-container>
  <ng-template #externalLink>
    <a [href]="pagelet.stringParam('Link')" [attr.title]="pagelet.stringParam('LinkTitle')" class="teaser-link">
      <ng-container *ngTemplateOutlet="image"></ng-container>
    </a>
  </ng-template>
</ng-container>

<ng-template #image>
  <div (deferLoad)="showImage = true" class="defer-load">
    <img
      *ngIf="showImage"
      [src]="pagelet.stringParam('Image')"
      [attr.alt]="pagelet.stringParam('AlternateText')"
      [ngClass]="pagelet.stringParam('CSSClass')"
    />
  </div>
</ng-template>

<!-- TODO: (CMS Component Configuration)
  Picture
-->
