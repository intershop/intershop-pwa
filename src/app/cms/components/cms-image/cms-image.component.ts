import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { ContentViewHelper } from 'ish-core/models/content-view/content-view.helper';
import { ContentPageletView } from 'ish-core/models/content-view/content-views';
import { CMSComponent } from '../../models/cms-component/cms-component.model';

@Component({
  selector: 'ish-cms-image',
  templateUrl: './cms-image.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CMSImageComponent implements CMSComponent {
  @Input() pagelet: ContentPageletView;

  routerLink = ContentViewHelper.getRouterLink;
}
