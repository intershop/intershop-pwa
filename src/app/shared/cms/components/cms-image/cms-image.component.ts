import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { ContentViewHelper } from 'ish-core/models/content-view/content-view.helper';
import { ContentPageletView } from 'ish-core/models/content-view/content-view.model';
import { CMSComponent } from 'ish-shared/cms/models/cms-component/cms-component.model';

@Component({
  selector: 'ish-cms-image',
  templateUrl: './cms-image.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CMSImageComponent implements CMSComponent {
  @Input() pagelet: ContentPageletView;

  isRouterLink = ContentViewHelper.isRouterLink;
  routerLink = ContentViewHelper.getRouterLink;

  /**
   * deferred loading flag
   */
  showImage = false;
}
