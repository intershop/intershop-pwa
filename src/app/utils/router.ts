import { ActivatedRoute, ChildActivationEnd, Router } from '@angular/router';
import { merge, Observable } from 'rxjs';
import { distinctUntilChanged, filter, pluck } from 'rxjs/operators';

/**
 * resolves data from child routes
 */
export function resolveChildRouteData<T>(route: ActivatedRoute, router: Router, key: string): Observable<T> {
  return merge(
    // get first data from injected activated route
    route.firstChild.data.pipe(pluck(key)),
    // listen to router events to get other child updates
    router.events.pipe(
      filter(event => event instanceof ChildActivationEnd),
      pluck('snapshot', 'data', key),
      filter((x: T) => !!x)
    )
  ).pipe(distinctUntilChanged());
}
