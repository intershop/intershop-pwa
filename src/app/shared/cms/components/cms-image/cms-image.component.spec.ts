import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { createContentPageletView } from 'ish-core/models/content-view/content-view.model';

import { CMSImageComponent } from './cms-image.component';

describe('Cms Image Component', () => {
  let component: CMSImageComponent;
  let fixture: ComponentFixture<CMSImageComponent>;
  let element: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [CMSImageComponent],
      imports: [RouterTestingModule],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CMSImageComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    const pagelet = {
      definitionQualifiedName: 'fq',
      id: 'id',
      displayName: 'name',
      domain: 'domain',
      configurationParameters: {
        Image: 'http://example.com/foo/bar.png',
        AlternateText: 'foo',
        CSSClass: 'foo',
        Link: 'http://example.com',
        LinkTitle: 'bar',
      },
    };
    component.pagelet = createContentPageletView(pagelet);
    component.showImage = true;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchInlineSnapshot(`
      <a class="teaser-link" href="http://example.com" title="bar"
        ><div class="defer-load">
          <img
            class="foo"
            ng-reflect-ng-class="foo"
            src="http://example.com/foo/bar.png"
            alt="foo"
          /></div
      ></a>
    `);
  });
});
