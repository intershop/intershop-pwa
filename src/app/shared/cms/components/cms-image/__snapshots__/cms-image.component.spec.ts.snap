// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cms Image Component should be created 1`] = `
<a class="teaser-link" href="http://example.com" title="bar"
  ><img class="foo" ng-reflect-ng-class="foo" src="http://example.com/foo/bar.png" alt="foo"
/></a>
`;
